cmd/virt-handler/virt-handler.go

```diff
@@ -348,11 +357,51 @@ func (app *virtHandlerApp) Run() {
 
 	go vmController.Run(10, stop)
 
+	doneCh := make(chan string)
+	defer close(doneCh)
+
 	errCh := make(chan error)
 	go app.runServer(errCh, consoleHandler, lifecycleHandler)
 
-	// wait for one of the servers to exit
-	fmt.Println(<-errCh)
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt,
+		syscall.SIGHUP,
+		syscall.SIGINT,
+		syscall.SIGTERM,
+		syscall.SIGQUIT,
+	)
+	// start graceful shutdown handler
+	go func() {
+		connectionInterval := 10 * time.Second
+		connectionTimeout := time.Duration(app.gracefulShutdownSeconds) * time.Second
+
+		s := <-c
+		log.Log.Infof("Received signal %s, initiating graceful shutdown", s.String())
+
```

Comment:
I thinnk we should be caling `migratoinProxy.GracefulShutdown()` here? I'm not sure how `m.isShuttingDown` is set to true otherwise, I might be missing something though?

Reply:
oh yeah... ha you're totally right. I made the function and forgot to call it. great catch

Reply:
fixed this a few lines below